radis/tools/slit.py

```diff
@@ -583,16 +635,50 @@ def convolve_with_slit(
     # --------------
 
     # Assert slit function is thin enough
-    try:
-        assert abs(w[-1] - w[0]) > abs(w_slit[-1] - w_slit[0])
-    except AssertionError:
-        raise AssertionError(
-            "Slit function is broader ({0:.1f}nm) than spectrum".format(
-                abs(w_slit[-1] - w_slit[0])
-            )
-            + " ({0:.1f}nm). No valid range.".format(abs(w[-1] - w[0]))
+
+    def lin_interp(x, y, i, half):
+        return x[i] + (x[i+1] - x[i]) * ((half - y[i]) / (y[i+1] - y[i]))
+
+    def half_max_x(x, y):
+        half = max(y)/2.0
+        signs = np.sign(np.add(y, -half))
+        zero_crossings = (signs[0:-2] != signs[1:-1])
+        zero_crossings_i = np.where(zero_crossings)[0]
+        return [lin_interp(x, y, zero_crossings_i[0], half),
+                lin_interp(x, y, zero_crossings_i[1], half)]
+    def fwhm(x,y):
+        hmx = half_max_x(x,y)
+        return hmx[1] - hmx[0]
+
+    w_range = w[-1] - w[0]
+    w_slit_range = w_slit[-1] - w_slit[0]
+    slit_FWHM = fwhm(w_slit, I_slit)
+
+    if np.abs(w_slit_range) >10*slit_FWHM:
+        warn(
+            f"FWHM >> spectral range!\n" \
+          + f"Slit Function is provided with a spectral range {w_slit_range:.1f} nm that is much " \
+          + f"wider than its estimated FWHM {slit_FWHM:.1f}. We recommend truncating the input " \
+          + f"slit function. See radis documentation about convolve_with_slit for more information."
         )
 
+    if np.abs(w_range) < np.abs(w_slit_range):
+        if mode == "valid":
+            raise AssertionError(
+                f"Slit spectral range > Spectrum spectral range and mode is set to valid!\n" \
+              + f"Slit function is provided with a spectral range ({w_slit_range:.1f} nm) larger than " \
+              + f"the spectral range of the spectrum ({w_range:.1f} nm). The output spectrum is therefore empty! " \
+              + f"Set mode to same in this case. However, we recommend truncating the input slit function. "
+              + f"See radis documentation about convolve_with_slit for more information."
```

Comment:
```suggestion
              f"Slit function is provided with a spectral range ({w_slit_range:.1f} {waveunit}) larger than " \
              + f"the spectral range of the spectrum ({w_range:.1f} {waveunit}) : the output spectrum will therefore be empty " \
              + f"as boundary effects of the convolution are automatically discarded. If you still want to apply the slit "
              + f"despite potential boundary effects, set `mode='same'`. However, we recommend truncating the input slit function if possible, or compute the spectrum on a larger spectral range. "
              + f"See radis documentation about convolve_with_slit for more information."
```

Note : spectra can be in wavenumbers and not in nm. 
Slit & spectrum are always given in the same waveunit, that's ensured by the Spectrum.apply_slit method.